proposals/NNNN-multi-pattern-catch-clauses.md

```diff
@@ -0,0 +1,128 @@
+# Multi-Pattern Catch Clauses
+
+* Proposal: [SE-NNNN](NNNN-multi-pattern-catch-clauses.md)
+* Authors: [Owen Voorhees](https://github.com/owenv)
+* Review Manager: TBD
+* Status: **Awaiting review**
+* Implementation:
+    - [apple/swift#27776](https://github.com/apple/swift/pull/27776)
+
+## Introduction
+
+Currently, each catch clause in a do-catch statement may only contain a single pattern and where clause. This is inconsistent with the behavior of cases in switch statements, which provide similar functionality. It also makes some error handling patterns awkward to express. This proposal extends the grammar of catch clauses to support a comma-separated list of patterns (with optional where clauses), resolving this inconsistency.
+
+Swift-evolution thread: [Thread](https://forums.swift.org/t/multi-pattern-and-conditionally-compiled-catch-clauses/30246)
+
+## Motivation
+
+Currently, Swift only allows up to one pattern and where clause for each catch clause in a do-catch statement, so the following code snippet is not allowed:
+
+```swift
+do {
+  try performTask()
+} catch TaskError.someRecoverableError {    // OK
+  recover()
+} catch TaskError.someFailure(let msg),
+        TaskError.anotherFailure(let msg) { // Not currently valid
+  showMessage(msg)
+}
+```
+
+Because the above snippet is not valid today, developers frequently end up duplicating code between catch clauses, or writing something like the following instead:
+
+```swift
+do {
+  try performTask()
+} catch {
```

Comment:
Could the `default: break` be removed from the switch?

```suggestion
} catch let error as TaskError {
```

Reply:
👍 , I think a fairer assessment of the current situation